src/docfx/lib/markdown/IncludeExtension.cs

```diff
@@ -64,8 +64,12 @@ private static void RemoveIncludeRenderer(IMarkdownRenderer renderer)
         private static void ExpandInclusionBlock(
             MarkdownContext context, InclusionBlock inclusionBlock, MarkdownPipeline pipeline, MarkdownPipeline inlinePipeline, ErrorBuilder errors)
         {
-            if (!string.IsNullOrEmpty(inclusionBlock.IncludedFilePath)
-                && !inclusionBlock.IncludedFilePath.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+            if (string.IsNullOrEmpty(inclusionBlock.IncludedFilePath))
```

Comment:
I am wondering, would it be easier to log the error in `linkResolver` directly?
https://github.com/dotnet/docfx/blob/ee2c1aa238e40824789c2b829411f2d1913fe63f/src/docfx/build/link/LinkResolver.cs#L60

Reply:
Current `TryResolveFile()` will resolve the empty link as SelfBookmark, we need identity the link type(include/markdown link) if we want to write `IncludeNotFound` error here.

Before go into `TryResolveFile()`, in `ExpandInclusionBlock()`, we have already check if the link is empty or not when validating `IncludeInvalid`, I think we can write `IncludeNotFound` here before the process `ReadFile() => ResolveContent() => TryResolveFile()`

Reply:
Got your point, makes sense to me now.